embedded/tbtree/tbtree.go

```diff
@@ -203,27 +213,178 @@ func Open(path string, opts *Options) (*TBtree, error) {
 		}
 	}
 
-	appendableOpts.WithFileExt("n")
-	nLog, err := appFactory(path, "nodes", appendableOpts)
+	appendableOpts.WithFileExt("hx")
+	hLog, err := appFactory(path, historyFolder, appendableOpts)
 	if err != nil {
 		return nil, err
 	}
 
-	appendableOpts.WithFileExt("hx")
-	hLog, err := appFactory(path, "history", appendableOpts)
+	// If compaction was not fully completed, a valid or partially written full snapshot may be there
+	snapIDs, err := recoverFullSnapshots(path, commitFolder, opts.log)
+	if err != nil {
+		return nil, err
+	}
+
+	// Try snapshots from newest to older
+	for i := len(snapIDs); i > 0; i-- {
+		snapID := snapIDs[i-1]
+
+		nFolder := snapFolder(nodesFolder, snapID)
+		cFolder := snapFolder(commitFolder, snapID)
+
+		snapPath := filepath.Join(path, cFolder)
+
+		opts.log.Infof("Reading snapshot at '%s'...", snapPath)
+
+		appendableOpts.WithFileExt("n")
+		nLog, err := appFactory(path, nFolder, appendableOpts)
+		if err != nil {
+			return nil, err
+		}
+
+		appendableOpts.WithFileExt("ri")
+		cLog, err := appFactory(path, cFolder, appendableOpts)
+		if err != nil {
+			return nil, err
+		}
+
+		// TODO: semantic validation and further ammendment procedures may be done instead of a full initialization
+		t, err := OpenWith(path, nLog, hLog, cLog, opts)
+		if err != nil {
+			opts.log.Infof("Error reading snapshot at '%s'. %v", snapPath, err)
+
+			err = nLog.Close()
+			if err != nil {
+				return nil, err
+			}
+
+			err = cLog.Close()
+			if err != nil {
+				return nil, err
+			}
+
+			err = discardSnapshots(path, snapIDs[i-1:i], opts.log)
+			if err != nil {
+				return nil, err
```

Comment:
probably a warning would work, but if there is an impossibility to discard previous snapshots it may end-up in a out of disk space error over time. @SimoneLazzaris?

Reply:
It all depends in how likely is that to happen and what is needed to solve the issue. If the warning has enough information, since (AFAIU) the only possible errors are coming from the "os" module, I think that we may want to risk it to preserve immudb operability